### PR TITLE
docs: update developer readme instruction on python env creation

### LINKED ISSUE #1690: [BUG] Update development document to add statement that it does not support development on Windows?
### SynapseML version

0.10.1

### System information

Windows OS with Spark installed.

### Describe the problem

On Windows OS, follow the development document and tried to create a virtual environment.
https://microsoft.github.io/SynapseML/docs/reference/developer-readme/

> conda env create -f environment.yml

It failed on the step of installing package "horovod==0.25.0", which was added 3 months ago within this commit: https://github.com/microsoft/SynapseML/commit/44c8ed5239dd7b7f43295865ff5b0caa87d40ab6

It appears that Horovod does not support Windows. https://github.com/horovod/horovod/blob/master/docs/install.rst

With this change, can you update the development readme to clarify that Python package building does not work on Windows OS?

### Code to reproduce issue

```
conda env create -f environment.yml
```

### Other info / logs

_No response_

### What component(s) does this bug affect?

- [ ] `area/cognitive`: Cognitive project
- [ ] `area/core`: Core project
- [ ] `area/deep-learning`: DeepLearning project
- [ ] `area/lightgbm`: Lightgbm project
- [ ] `area/opencv`: Opencv project
- [ ] `area/vw`: VW project
- [ ] `area/website`: Website
- [ ] `area/build`: Project build system
- [ ] `area/notebooks`: Samples under notebooks folder
- [ ] `area/docker`: Docker usage
- [ ] `area/models`: models related issue

### What language(s) does this bug affect?

- [ ] `language/scala`: Scala source code
- [ ] `language/python`: Pyspark APIs
- [ ] `language/r`: R APIs
- [ ] `language/csharp`: .NET APIs
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [ ] `integrations/synapse`: Azure Synapse integrations
- [ ] `integrations/azureml`: Azure ML integrations
- [ ] `integrations/databricks`: Databricks integrations

### PR DESCRIPTION
**Pull request by serena-ruan**
## Related Issues/PRs

Close #1690 

## What changes are proposed in this pull request?

_Briefly describe the changes included in this Pull Request._

## How is this patch tested?

- [ ] I have written tests (not required for typo or doc fix) and confirmed the proposed feature/bug-fix/change works.

## Does this PR change any dependencies?

- [ ] No. You can skip this section.
- [ ] Yes. Make sure the dependencies are resolved correctly, and list changes here.

## Does this PR add a new feature? If so, have you added samples on website?

- [ ] No. You can skip this section.
- [ ] Yes. Make sure you have added samples following below steps.

1. Find the corresponding markdown file for your new feature in `website/docs/documentation` folder.
   Make sure you choose the correct class `estimators/transformers` and namespace.
2. Follow the pattern in markdown file and add another section for your new API, including pyspark, scala (and .NET potentially) samples.
3. Make sure the `DocTable` points to correct API link.
4. Navigate to website folder, and run `yarn run start` to make sure the website renders correctly.
5. Don't forget to add `<!--pytest-codeblocks:cont-->` before each python code blocks to enable auto-tests for python samples.
6. Make sure the `WebsiteSamplesTests` job pass in the pipeline.

### CODE REVIEW

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards

**A minimum Acrolinx score of 80 is required.**

Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology.

| Article | Acrolinx<br>score | Word and<br>Phrases Score | Correctness<br>Score | Scorecard | Processed |
|---------|:--------------:|:--------------------:|:------:|:---------:|:---------:|
| [website/docs/reference/developer-readme.md](https://github.com/microsoft/SynapseML/blob/6d11c0d717850deea0f6d8d59a96632202c075c2/website%2Fdocs%2Freference%2Fdeveloper-readme.md) |   |   |   | [link]( ) | [:warning:](# "Could not check with Acrolinx&#10;Please close and reopen pull request") |
| [website/versioned_docs/version-0.10.1/reference/developer-readme.md](https://github.com/microsoft/SynapseML/blob/6d11c0d717850deea0f6d8d59a96632202c075c2/website%2Fversioned_docs%2Fversion-0.10.1%2Freference%2Fdeveloper-readme.md) |   |   |   | [link]( ) | [:warning:](# "Could not check with Acrolinx&#10;Please close and reopen pull request") |

**More information about Acrolinx**

- [Install Acrolinx locally for VSCode](https://review.docs.microsoft.com/help/contribute/contribute-acrolinx-vscode)
- [False positives or issues](https://aka.ms/acrolinxbug)
- [Troubleshooting issues with Acrolinx](https://review.docs.microsoft.com/help/contribute/acrolinx-error-messages)

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards

**A minimum Acrolinx score of 80 is required.**

Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology.

| Article | Acrolinx<br>score | Word and<br>Phrases Score | Correctness<br>Score | Scorecard | Processed |
|---------|:--------------:|:--------------------:|:------:|:---------:|:---------:|
| [website/docs/reference/developer-readme.md](https://github.com/microsoft/SynapseML/blob/6d11c0d717850deea0f6d8d59a96632202c075c2/website%2Fdocs%2Freference%2Fdeveloper-readme.md) |   |   |   | [link]( ) | [:warning:](# "Could not check with Acrolinx&#10;Please close and reopen pull request") |
| [website/versioned_docs/version-0.10.1/reference/developer-readme.md](https://github.com/microsoft/SynapseML/blob/6d11c0d717850deea0f6d8d59a96632202c075c2/website%2Fversioned_docs%2Fversion-0.10.1%2Freference%2Fdeveloper-readme.md) |   |   |   | [link]( ) | [:warning:](# "Could not check with Acrolinx&#10;Please close and reopen pull request") |

**More information about Acrolinx**

- [Install Acrolinx locally for VSCode](https://review.docs.microsoft.com/help/contribute/contribute-acrolinx-vscode)
- [False positives or issues](https://aka.ms/acrolinxbug)
- [Troubleshooting issues with Acrolinx](https://review.docs.microsoft.com/help/contribute/acrolinx-error-messages)

File: website/docs/reference/developer-readme.md [hunk 1]
```diff
@@ -30,6 +30,11 @@ description: SynapseML Development Setup
     - Install [Miniconda](https://docs.conda.io/en/latest/miniconda.html)
     - Note: if you want to run conda commands from IntelliJ, you may need to select the option to add conda to PATH during installation.
     - Activate the `synapseml` conda environment by running `conda env create -f environment.yml` from the `synapseml` directory.
+    :::note
+    If you're using a Windows machine, please remove
+    `horovod==0.25.0` requirement in the environment.yml file, because horovod installation only
```
Comment (dylanw-oss) [inline website/docs/reference/developer-readme.md:35]: If horovod version is upgraded in the future, this needs update again, better to just mention that remove `horovod` requirement from environment.yml

**Review by dylanw-oss — APPROVED** [review]

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards

**A minimum Acrolinx score of 80 is required.**

Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology.

| Article | Acrolinx<br>score | Word and<br>Phrases Score | Correctness<br>Score | Scorecard | Processed |
|---------|:--------------:|:--------------------:|:------:|:---------:|:---------:|
| [website/docs/reference/developer-readme.md](https://github.com/microsoft/SynapseML/blob/5bc3cc315362d24c7cdc187a210c1e72b5662ad3/website%2Fdocs%2Freference%2Fdeveloper-readme.md) |   |   |   | [link]( ) | [:warning:](# "Could not check with Acrolinx&#10;Please close and reopen pull request") |
| [website/versioned_docs/version-0.10.1/reference/developer-readme.md](https://github.com/microsoft/SynapseML/blob/5bc3cc315362d24c7cdc187a210c1e72b5662ad3/website%2Fversioned_docs%2Fversion-0.10.1%2Freference%2Fdeveloper-readme.md) |   |   |   | [link]( ) | [:warning:](# "Could not check with Acrolinx&#10;Please close and reopen pull request") |

**More information about Acrolinx**

- [Install Acrolinx locally for VSCode](https://review.docs.microsoft.com/help/contribute/contribute-acrolinx-vscode)
- [False positives or issues](https://aka.ms/acrolinxbug)
- [Troubleshooting issues with Acrolinx](https://review.docs.microsoft.com/help/contribute/acrolinx-error-messages)

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards

**A minimum Acrolinx score of 80 is required.**

Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology.

| Article | Acrolinx<br>score | Word and<br>Phrases Score | Correctness<br>Score | Scorecard | Processed |
|---------|:--------------:|:--------------------:|:------:|:---------:|:---------:|
| [website/docs/reference/developer-readme.md](https://github.com/microsoft/SynapseML/blob/fc6090544273dd1afc477d166ba812dc30d46a82/website%2Fdocs%2Freference%2Fdeveloper-readme.md) |   |   |   | [link]( ) | [:warning:](# "Could not check with Acrolinx&#10;Please close and reopen pull request") |
| [website/versioned_docs/version-0.10.1/reference/developer-readme.md](https://github.com/microsoft/SynapseML/blob/fc6090544273dd1afc477d166ba812dc30d46a82/website%2Fversioned_docs%2Fversion-0.10.1%2Freference%2Fdeveloper-readme.md) |   |   |   | [link]( ) | [:warning:](# "Could not check with Acrolinx&#10;Please close and reopen pull request") |

**More information about Acrolinx**

- [Install Acrolinx locally for VSCode](https://review.docs.microsoft.com/help/contribute/contribute-acrolinx-vscode)
- [False positives or issues](https://aka.ms/acrolinxbug)
- [Troubleshooting issues with Acrolinx](https://review.docs.microsoft.com/help/contribute/acrolinx-error-messages)

**Pull request merged**